### PR TITLE
Implemented the remove so that it takes into account of NB links

### LINKED ISSUE #478: Properly Implement Delete on links/intentions/actors

### PR DESCRIPTION
**Pull request by Domily11**

### CODE REVIEW

File: leaf-ui/js/onFunctions.js [hunk 1]
```diff
@@ -707,14 +685,16 @@ function clearInspector() {
...
-    for (var i = 0; i < localLinks.length; i++) {
-        if (localLinks[i].prop("link-type") == 'NBT' || localLinks[i].prop("link-type") == 'NBD') {
-            if (localLinks[i].getSourceElement().prop("id") == node["id"] || localLinks[i].getTargetElement().prop("id") == node["id"]) {
-                num += 1;
+    var localLinks = graph.getConnectedLinks(element);
+
+    if(localLinks != null){
+        for (var i = 0; i < localLinks.length; i++) {
+            if (localLinks[i].get('link').get("link-type") == 'NBT' || localLinks[i].get('link').get("link-type") == 'NBD') {
+             //   if (localLinks[i].getSourceElement().prop("id") == node["id"] || localLinks[i].getTargetElement().prop("id") == node["id"]) {
```
Comment (Manarnz) [inline leaf-ui/js/onFunctions.js]: delete line as well as line 697

File: leaf-ui/rappid-extensions/LinkInspector.js [hunk 2]
```diff
@@ -170,11 +170,13 @@ var LinkInspector = Backbone.View.extend({
...
-            
-            // targetIntention.changeInitialSatValue('0000');
-            // targetIntention.dynamicFunction.stringDynVis = 'NB';
+            source.get('intention').get('evolvingFunction').set('functionSegList', []);
+            source.get('intention').get('evolvingFunction').set('type', 'NB');
+            source.get('intention').getUserEvaluationBBM(0).set('assignedEvidencePair', '(no value)');
+
+            target.get('intention').get('evolvingFunction').set('functionSegList', []);
+            target.get('intention').get('evolvingFunction').set('type', 'NB');
+            target.get('intention').getUserEvaluationBBM(0).set('assignedEvidencePair', '(no value)')
```
Comment (Manarnz) [inline leaf-ui/rappid-extensions/LinkInspector.js]: Add `;`

**Review by Manarnz — CHANGES_REQUESTED** [review]
Whenever a change is made in the paper the initial sat value disappears. Overall, great job!

File: leaf-ui/js/onFunctions.js [hunk 3]
```diff
@@ -188,52 +188,30 @@ graph.on('remove', function (cell) {
...
-    //     }
-    //     if (target !== null && !checkForMultipleNB(target)) {
-    //         target.attrs(".funcvalue/text", "");
-    //     }
-    // }
+    else if (cell.isLink() && (cell.get('link').get("linkType") == 'NBT' || cell.get('link').get("linkType") == 'NBD')) {
+        // Verify if is a Not both type. If it is remove labels from source and target node
+        var source = graph.getCell(cell.get('source').id);
+        var target = graph.getCell(cell.get('target').id);
+        //Verify if it is possible to remove the NB tag from source and target
```
Comment (Manarnz) [inline leaf-ui/js/onFunctions.js]: Add space between `//` and `Verify`

**Review by Manarnz — DISMISSED** [review]
Great job Cat!!

**Review by amgrubb — DISMISSED** [review]
Great job!

**Review by amgrubb — APPROVED** [review]

**Pull request merged**